scheduler/cmd/scheduler/main.go

```diff
@@ -341,3 +393,68 @@ func main() {
 
 	log.Info("Shutdown services")
 }
+
+type probe struct {
+	port       int
+	plaintText bool
+}
+
+type gRPCHealthProbes struct {
+	probes []probe
+}
+
+func initHealthProbe(tlsOptions *tls.TLSOptions, log *log.Logger, config gRPCHealthProbes, healthSrvPort int) (*health_probe.HTTPServer, error) {
+	manager := health_probe.NewManager()
+	for _, probe := range config.probes {
+		var cert *tls.CertificateStore
+		if !probe.plaintText {
+			cert = tlsOptions.Cert
+		}
+		if err := createGRPCHealthProbe(cert, probe.port, manager); err != nil {
```

Comment:
Because all the gRPC server sub-services are local, there was also the option of simply storing some readiness state in the associated data structures, without adding gRPC requests. However, perhaps your idea here is better from the perspective of testing those servers like an external client would. With readiness and liveness probes, we are adding almost 2 RPS background load to the system (beyond the keepalives), but I'm satisfied that the processing for each is minimal enough not to matter. Comparatively, the keepalive RPS grows with the number of Servers, and can become significant if you have hundreds of Servers.